scipy/stats/tests/test_rank.py

```diff
@@ -192,6 +192,52 @@ def test_size_0_axis(self, axis, method, dtype):
         assert_equal(r.shape, shape)
         assert_equal(r.dtype, dtype)
 
+    def test_nan_policy(self):
```

Comment:
```suggestion
    methods = ['average', 'min', 'max', 'dense', 'ordinal']

    @pytest.mark.parametrize('method', methods)
    def test_nan_policy(self, method):
```

I tried making this change (and using `method=method` in calls to `rankdata` below) and found that tests fail for all methods except 'average'. This is pretty clear from the logic. I'd suggest refactoring so that there is only one `return` statement (at the end).